pyproject.toml

```diff
@@ -58,7 +58,7 @@ pylibjpeg-libjpeg = {version = "*", optional = true}  # groups = ["user", "dev",
 immutables = {version = "*", optional = true}  # groups = ["user", "dev", "dicom"]
 
 cython = {version = "*", optional = true}  # groups = ["user", "dev"]
-pymssql = {version = "< 3.0.0", python = "< 3.8", optional = true}  # groups = ["user", "dev"]
+pymssql = {version = "*", optional = true}  # groups = ["user", "dev"]
```

Comment:
doesn't this imply that if someone already has pymssql installed, any version will be considered OK?
wouldn't it be better to require at least the current version?
Or is it enough to assume that if someone is starting fresh (e.g. the CI) they will get the latest and not have a "too old" version of Python?

Reply:
Best practices here is to have as loose as possible restrictions on a library package, only ever apply a version restriction if it is certainly known that a given version range won't work. For application deployment, pin all versions to a fixed exact version that has been tested.

Given we're on the library end, not the application end, it is better for us to be as open as possible so that we won't cause downstream "dependency hell".

Reply:
I should add here, I am choosing to pin Streamlit, as their API is still in development.

Reply:
With streamlit being as unstable as it is (not a dig, they are making great progress and have a very powerful model), and streamlit being what one might expect a physicist or developer in a clinical/research setting to want to use... this is going to get sticky.  It should be clearly documented for those using pymedphys as a development library/package

Reply:
> It should be clearly documented for those using pymedphys as a development library/package

Yup fair point. At the moment there actually isn't even any mention of the GUI or Streamlit in the docs. We essentially need a whole "GUI" section in the documentation which contains advice for users, advice for GUI creators, and some guidance on testing practices...

There's a lot to document in that space...

Reply:
No time like the present to create a nearly empty section (in another Issue/PR).
I'll own the sub-section on pseudonymisation GUI...

Reply:
> No time like the present to create a nearly empty section (in another Issue/PR).

Yup :) good call. Will do.